meshmode/array_context.py

```diff
@@ -206,6 +208,19 @@ def get_container_context(ary: ArrayContainer) -> Optional["ArrayContext"]:
     """
     return None
 
+
+def _get_container_context_via_attr(ary):
+    return ary.array_context
+
+
+def has_array_context_attribute(cls):
+    r"""A class decorator for :class:`ArrayContainer`\ s that registers an
+    implementation of :class:`get_container_context` that uses an
+    ``array_context`` attribute that must be present.
+    """
+    get_container_context.register(cls)(_get_container_context_via_attr)
+    return cls
```

Comment:
What do you think about just returning `getattr(ary, 'array_context', None)` by default? Then none of the existing registered implementations would be required (at the cost of that `getattr`).

Reply:
867ccf70132f95db1f141ce77e92545f200cd359